### PR TITLE
XML Tasks of Marian Zlatev - DOM

### PR DESCRIPTION
**Pull request by NewestUser**
Completed DOM task.

### CODE REVIEW

File: src/com/clouway/xml/XMLParser.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    return null;
+  }
+
+  /**
+   * Get a instance of the specified class name.
+   *
+   * @param clazzName name of class.
+   * @return instance of that class.
+   */
+  private Object getInstance(String clazzName) {
```
Comment (mgenov) [inline src/com/clouway/xml/XMLParser.java]: Seems coupled with the current package structure, so I suggest you to extract this method as interface and pass it when parser is created. So anyone will be able to adapt it to it's own package structure.

File: tests/com/clouway/xml/XMLParserTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.*;
+
+public class XMLParserTest {
+
+  @Test
+  public void parseXml() throws Exception {
+    XMLParser parser = new XMLParser();
+    Employee employee = (Employee) parser.createRoot(new FileInputStream(new File("xml/employee.xml")));
```
Comment (mgenov) [inline tests/com/clouway/xml/XMLParserTest.java]: Please ensure that typecast is not required.

File: src/com/clouway/xml/XMLParser.java (hunk 1 above)
Comment (NewestUser) [inline src/com/clouway/xml/XMLParser.java]: Do I have to extract the entire method getInstance() with a interface or just create a interface that can return desired package name. 
Like:
public Interface Destination{
   public String getDestination();
}

Comment (mgenov) [inline src/com/clouway/xml/XMLParser.java]: The entire getInstance is ok.

On 1/30/15, MZlatev <notifications@github.com> wrote:
>> +      e.printStackTrace();
>> +    } catch (SAXException e) {
>> +      e.printStackTrace();
>> +    } catch (IOException e) {
>> +      e.printStackTrace();
>> +    }
>> +    return null;
>> +  }
>> +
>> +  /**
>> +   * Get a instance of the specified class name.
>> +   *
>> +   * @param clazzName name of class.
>> +   * @return instance of that class.
>> +   */
>> +  private Object getInstance(String clazzName) {
>
> Do I have to extract the entire method getInstance() with a interface or
> just create a interface that can return desired package name.
> Like:
> public Interface Destination{
>    public String getDestination();
> }
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/clouway-incubator/xml/pull/3/files#r23837733


-- 
Regards,
  Miroslav Genov

Comment (NewestUser) [inline src/com/clouway/xml/XMLParser.java]: getInstance() is extracted in a separate class called Instantiator.

**Comment by mgenov** [comment]
LGTM, but please merge first SAX implementation, then you can rebease this branch and merge it with the upstream.

**Pull request closed without merging**